core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatEntry.scala

```diff
@@ -461,14 +461,13 @@ class ColumnFormatValue extends SerializedDiskBuffer
       val perfStats = getCachePerfStats(context)
       val startDecompression = perfStats.startDecompression()
       val decompressed = CompressionUtils.codecDecompress(buffer, allocator, position, -typeId)
-      val isManagedDirect = allocator.isManagedDirect
       try {
         // update decompression stats
         perfStats.endDecompression(startDecompression)
         val newValue = copy(decompressed, isCompressed = false, changeOwnerToStorage = false)
```

Comment:
No decompression is required in either case because iterator needs to read the value (in generated ColumnTableScan code). The check (!isDirect || this.refCount <= 2) says whether the value stored in region can also be replaced with decompressed value or not. If yes, then this object will be changed and returned. If not, then a new ColumnFormatValue will be created on-the-fly, filled with decompressed buffer and returned.

The only case where decompression can be skipped is for "FetchRequest.DECOMPRESS_IF_IN_MEMORY" when "onlyIfStored" is passed is true to this method and handled separately before this block. That case is when value is being fetched for being sent over to a smart connector client and in that case decompression can be skipped altogether.

Reply:
Yeah. Got it.

Reply:
Thanks